File: app/src/main/java/com/tianma/tweaks/miui/xp/hook/launcher/MiuiLauncherHook.java
package com.tianma.tweaks.miui.xp.hook.launcher;

import com.tianma.tweaks.miui.data.sp.XPrefContainer;
import com.tianma.tweaks.miui.utils.XLogKt;
import com.tianma.tweaks.miui.utils.rom.MiuiUtils;
import com.tianma.tweaks.miui.xp.hook.BaseHook;

import de.robv.android.xposed.callbacks.XC_LoadPackage;

public class MiuiLauncherHook extends BaseHook {

    public static final String PACKAGE_NAME = "com.miui.home";

    public MiuiLauncherHook() {
    }

    @Override
    public void onLoadPackage(XC_LoadPackage.LoadPackageParam lpparam) throws Throwable {
        if (PACKAGE_NAME.equals(lpparam.packageName)) {
            XLogKt.logI("Hooking MIUI Launcher...");

            ClassLoader classLoader = lpparam.classLoader;
            if (XPrefContainer.getMainSwitchEnable()) {
                if(!MiuiUtils.isMiui()) {
                    return;
                }
                new WorkSpaceHook(classLoader).startHook();
            }

        }
    }
}

File: app/src/main/java/com/tianma/tweaks/miui/xp/hook/launcher/MiuiLauncherHook.kt
package com.tianma.tweaks.miui.xp.hook.launcher

import com.tianma.tweaks.miui.data.sp.XPrefContainer.mainSwitchEnable
import com.tianma.tweaks.miui.utils.logI
import com.tianma.tweaks.miui.utils.rom.MiuiUtils
import com.tianma.tweaks.miui.xp.hook.BaseHook
import de.robv.android.xposed.callbacks.XC_LoadPackage.LoadPackageParam

class MiuiLauncherHook : BaseHook() {

    companion object {
        const val PACKAGE_NAME = "com.miui.home"
    }

    @Throws(Throwable::class)
    override fun onLoadPackage(lpparam: LoadPackageParam) {
        if (PACKAGE_NAME == lpparam.packageName) {
            logI("Hooking MIUI Launcher...")
            val classLoader = lpparam.classLoader
            if (mainSwitchEnable) {
                if (!MiuiUtils.isMiui()) {
                    return
                }
                WorkSpaceHook(classLoader).startHook()
            }
        }
    }
}